File: app/src/main/java/io/eodc/planit/model/AssignmentListViewModel.java
package io.eodc.planit.model;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import org.joda.time.DateTime;

import java.util.List;

import io.eodc.planit.db.Assignment;
import io.eodc.planit.db.AssignmentDao;
import io.eodc.planit.db.PlannerDatabase;

public class AssignmentListViewModel extends AndroidViewModel {
    private AssignmentDao mAssignmentDao;

    public AssignmentListViewModel(@NonNull Application application) {
        super(application);
        mAssignmentDao = PlannerDatabase.getInstance(application).assignmentDao();
    }

    public LiveData<List<Assignment>> getAllAssignments() {
        return mAssignmentDao.getAllAssignments();
    }

    public LiveData<List<Assignment>> getOverdueAssignments(DateTime date) {
        return mAssignmentDao.getOverdueAssignments(date);
    }

    public LiveData<List<Assignment>> getAssignmentsDueOnDay(DateTime date) {
        return mAssignmentDao.getAssignmentsDueBetweenDates(date.withTimeAtStartOfDay(), date.plusDays(1).minusMillis(1));
    }

    public LiveData<List<Assignment>> getAssignmentsBetweenDates(DateTime from, DateTime to) {
        return mAssignmentDao.getAssignmentsDueBetweenDates(from, to);
    }

    public LiveData<List<Assignment>> getAssignmentsByClassId(int id) {
        return mAssignmentDao.getAssignmentsByClassId(id);
    }

    public LiveData<Assignment> getAssignmentById(int id) {
        return mAssignmentDao.getAssignmentById(id);
    }

    public void insertAssignments(Assignment... assignments) {
        mAssignmentDao.insertAssignments(assignments);
    }

    public void removeAssignments(Assignment... assignments) {
        mAssignmentDao.removeAssignment(assignments);
    }
}

File: app/src/main/java/io/eodc/planit/model/AssignmentListViewModel.kt
package io.eodc.planit.model

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import io.eodc.planit.db.Assignment
import io.eodc.planit.db.AssignmentDao
import io.eodc.planit.db.PlannerDatabase
import org.joda.time.DateTime

class AssignmentListViewModel(application: Application) : AndroidViewModel(application) {
    private val mAssignmentDao: AssignmentDao

    val allAssignments: LiveData<List<Assignment>>
        get() = mAssignmentDao.allAssignments

    init {
        mAssignmentDao = PlannerDatabase.getInstance(application)!!.assignmentDao()
    }

    fun getOverdueAssignments(date: DateTime): LiveData<List<Assignment>> {
        return mAssignmentDao.getOverdueAssignments(date)
    }

    fun getAssignmentsDueOnDay(date: DateTime): LiveData<List<Assignment>> {
        return mAssignmentDao.getAssignmentsDueBetweenDates(date.withTimeAtStartOfDay(), date.plusDays(1).minusMillis(1))
    }

    fun getAssignmentsBetweenDates(from: DateTime, to: DateTime): LiveData<List<Assignment>> {
        return mAssignmentDao.getAssignmentsDueBetweenDates(from, to)
    }

    fun getAssignmentsByClassId(id: Int): LiveData<List<Assignment>> {
        return mAssignmentDao.getAssignmentsByClassId(id)
    }

    fun getAssignmentById(id: Int): LiveData<Assignment> {
        return mAssignmentDao.getAssignmentById(id)
    }

    fun insertAssignments(vararg assignments: Assignment) {
        mAssignmentDao.insertAssignments(*assignments)
    }

    fun removeAssignments(vararg assignments: Assignment) {
        mAssignmentDao.removeAssignment(*assignments)
    }
}
